### PR TITLE
Pb-5

### PR DESCRIPTION
**Pull request by AR1988**

### CODE REVIEW

File: phonebook-ui/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
 
-const routes: Routes = [];
+const routes: Routes = [
+  {
+    path: '', component: AuthLayoutComponent, children: [
+      // { path: '', redirectTo: '/login', pathMatch: 'full' },
+      {path: '', redirectTo: 'user/register', pathMatch: 'full'},
+      {path: 'user/register', component: RegisterPageComponent},
+      {path: 'user/activation', component: ActivationPageComponent},
+      {path: 'user/pending/:token', component: PendingPageComponent},
```
Comment (sharf85) [inline phonebook-ui/src/app/app-routing.module.ts:16]: make the last two routes vice versa. Meaning, user/activation/token and user/pending

File: phonebook-ui/src/app/auth/pending-page/pending-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  confirmMessage: string;
+
+  ngOnInit(): void {
+    this.sendToken()
+  }
+
+  sendToken() {
+    const token = this.route.snapshot.paramMap.get('token');
+
+    const messageSuccess = 'Your email was successfully confirmed!';
```
Comment (sharf85) [inline phonebook-ui/src/app/auth/pending-page/pending-page.component.ts:28]: you can pull these two vars out off the method

File: phonebook-ui/src/app/auth/register-page/register-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+              private router: Router,
+              private route: ActivatedRoute) {
+  }
+
+  ngOnInit(): void {
+    this.form = new FormGroup({
+        email: new FormControl(null,
+          [
+            Validators.required,
+            Validators.pattern("^[a-z0-9._-]+@[a-z0-9.-]+\\.[a-z]{2,10}$")
```
Comment (sharf85) [inline phonebook-ui/src/app/auth/register-page/register-page.component.ts:33]: whats does mean double back slash?

File: phonebook-ui/src/app/common/service/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  constructor(private http: HttpClient) {
+  }
+
+  registerNewUser(user: User) {
+    const url = `${this.host}api/v1/registration`;
+    return this.http.post<User>(url, user);
+  }
+
+  sendRequestToRegisterConfirm(token: string) {
+    const url = `${this.host}api/v1/confirmation?token=${token}`;
```
Comment (sharf85) [inline phonebook-ui/src/app/common/service/user.service.ts:21]: all paths should be constants

File: phonebook-ui/src/app/common/service/user.service.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import {HttpClient} from "@angular/common/http";
+import {User} from "../model/user";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+
+  private host = 'http://localhost:8080/';
+  private registerUrl = `${this.host}api/v1/registration`;
```
Comment (sharf85) [inline phonebook-ui/src/app/common/service/user.service.ts:11]: api/v1/registration -> api/v1/user/registration
the same api/v1/user/confirmation

**Comment by igoscheff** [comment]
It is better to use a proxy to automatically add the URL of our API to all URIs starting with /api.

File: phonebook-ui/src/app/common/service/user.service.ts (hunk 5 above)
Comment (igoscheff) [reply phonebook-ui/src/app/common/service/user.service.ts]: I suggest "v1" to remove!

Comment (igoscheff) [reply phonebook-ui/src/app/common/service/user.service.ts]: otherwise, in version 2, we will change all links on the frontend and backend!!!

**Pull request merged**